### PR TITLE
Pull request for the GENTPF/GENTPJ models

### PR DESCRIPTION
**Pull request by Carlos-Alegre**
Pull Request for the ID-196 GENTPF/GENTPJ synchronous machine models

### CODE REVIEW

File: pages/models/generations/GENTPJ/index.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+## Model use, assumptions, validity domain and limitations
+
+The model is used to perform transient stability analysis of synchronous machines, providing a more accurate representation of the physical machine as it considers that there exist saturation in the iron of the machine, and it is applied to all the inductive terms simultaneously, differently to what the GENROU/GENSAL models with additive saturation consider. Following a recommendation note from the North-America Electrical Reliability Corporation (NERC) [[4]](4), the GENTPJ is overall recommended for all the synchronous machines as it recognizes the effect of the stator current in saturation. The GENROU, GENSAL and GENTPF (the later being exactly identical to the GENTPJ except an additive term in the saturation function) may induce significant error specially in simulations where reactive power support is considered.
```
Comment (GJCRESYM) [inline pages/models/generations/GENTPJ/index.md:18]: (NERC) [[4]](4) - link is not working 
image

**Review by GJCRESYM — CHANGES_REQUESTED** [review]

File: pages/models/generations/GENTPJ/index.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+The model assumptions are the following:
+
+* The three-phase system is balanced
+* The air gap between the stator and rotor is non-uniform, but the relationships between the flux linkages and currents are independent of the position of the rotor when expressed in the *dq0* reference frame.
+* The magnetic circuit is considered to be non-linear, considering saturation effects, and can be modeled using different saturation functions.
+* The saturation effects are applied to the subtransient reactances.
+
+The GENTPJ model is suitable for EMT as it models with detail the non-linearities and the fast dynamics of the machine. However, the saturation effects are only applied to the subtransient reactances. Newer models consider saturation over even more elements of the machine, improving GENTPF and GENTPJ models. GENTPJ will be deprecated in December 2024, and substituted by the [GENQEC](../GENQEC/) [[5]](#5), although this later model is still in testing process [[6]](#6).
```
Comment (GJCRESYM) [inline pages/models/generations/GENTPJ/index.md:27]: [GENQEC](../GENQEC/) - link is not working

File: pages/models/generations/GENTPJ/index.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+Figure 1: Block diagram of the GENTPJ model
+</div>
+<br>
+
+## Open source implementations
+
+No open-source implementations were found.
+
+
+## Table of references
```
Comment (GJCRESYM) [inline pages/models/generations/GENTPJ/index.md:213]: Please refer to the model template and add as much as information possible to tables of references

http://127.0.0.1:4000/pages/templates/modelTemplate.html

image

**Review by GJCRESYM — CHANGES_REQUESTED** [review]

File: pages/models/generations/GENTPJ/index.md (hunk 1 above)
Comment (Carlos-Alegre) [reply pages/models/generations/GENTPJ/index.md:18]: corrected

File: pages/models/generations/GENTPJ/index.md (hunk 2 above)
Comment (Carlos-Alegre) [reply pages/models/generations/GENTPJ/index.md:27]: This one will work once the GENQEC model is uploaded

File: pages/models/generations/GENTPJ/index.md (hunk 3 above)
Comment (Carlos-Alegre) [reply pages/models/generations/GENTPJ/index.md:213]: Improved the references

**Review by GJCRESYM — APPROVED** [review]

**Comment by GJCRESYM** [comment]
@Carlos-Alegre : Change the pull requests to dev instead of "main"

**Review by GJCRESYM — APPROVED** [review]

**Review by GJCRESYM — APPROVED** [review]

**Pull request merged**